homeassistant/components/sensor/velo_antwerpen.py

```diff
@@ -0,0 +1,107 @@
+"""
+Shows the available amount of public city bikes for Velo Antwerpen.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.velo_antwerpen/
+"""
+
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Velo'
+DEFAULT_ICON = 'mdi:bike'
+DEFAULT_UNIT = 'bikes'
+
+CONF_STATION_ID = 'station_id'
+
+RESOURCE_URL = "https://www.velo-antwerpen.be/availability_map/getJsonObject"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Velo sensor."""
+    name = config.get(CONF_NAME)
+    station = config.get(CONF_STATION_ID)
+
+    add_entities([VeloSensor(name, station)], True)
+
+
+class VeloSensor(Entity):
+    """Get the available amount of bikes and set the selected station as attributes"""
+
+    def __init__(self, name, station):
+        """Initialize the Velo sensor."""
+        self._name = name
+        self._station = station
+        self._state = None
+        self._station_data = self.make_request()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return DEFAULT_UNIT
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return DEFAULT_ICON
+
+    @property
+    def station(self):
+        """Return the station id"""
+        return self._station
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            "Station ID": self._station,
+            "Station": self._station_data["address"],
+            ATTR_ATTRIBUTION: "https://www.velo-antwerpen.be/",
+        }
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    def make_request(self):
```

Comment:
Protocol handling is not allowed inside Home Assistant (any more).

This method, the `RESOURCE_URL` constant and probably the attribution text must thus be moved to a separate library published on PyPI.

It would be nice if that library would massage the JSON into a proper Python object but that is not a requirement.

Reply:
Hmm, this troubles me. I barely know enough Python to get by and I was already glad I could achieve the changes in this PR. This is largely due to how well Home Assistant helps me with out of the box pieces of code and linting.

What you're now essentially asking me to do as a first/second time contributor:
- Learn Python
- Learn how to write modules
- Learn how to publish this module to PyPI
- Learn how to set up tools like flake8, autopep for my module, which came out of the box for this change.

Since this would mean I would create a new module on PyPI that would also mean I'd have to maintain this module in the PyPI ecosystem.

I find it hard to understand you would rather see me create a module with code that lives online, outside of Home Assistant (where I could possibly do other stuff) than this simple GET call to an API which I think is easy to review (correct me if I'm wrong).

I really do love Home Assistant and was super excited about this PR and being able to add a new platform with my limited knowledge and Python skills. Now I'm kind of sad that I would have to revert all this again.

If this really is a required change and a no-go for merging could you let me know in a comment below this one? Then I will close this PR and unfortunately release it as a custom_component on my own Github or on the @custom-components account.

Thanks for taking the time to review my code.

Reply:
I'm sorry; I know it is discouraging but this is the work that has to be put in. I can understand if you don't think it is worth the effort.

It has turned out to be too hard to draw a line between simple enough and too complex, especially because simple implementations often grow error handling and new features. So the current rule is that no communication at all is allowed for new contributions.

Reply:
Ok, then I'll close this PR. Too bad, but if those are the rules, I'll comply.